hail/src/main/scala/is/hail/annotations/Region.scala

```diff
@@ -23,12 +27,15 @@ object Region {
 //    within-Region references to/from absolute addresses.
 
 final class Region private (empty: Boolean) extends NativeBase() {
-  def this() { this(false) }
-  @native def nativeCtor(p: RegionPool): Unit
+  def this(blockSize: Int = 64 * 1024) {
+    this(false)
+    nativeCtor(RegionPool.get, blockSize)
+  }
+  @native def nativeCtor(p: RegionPool, blockSize: Int): Unit
   @native def initEmpty(): Unit
   @native def nativeClearRegion(): Unit
 
-  if (empty) initEmpty() else nativeCtor(RegionPool.get)
+  if (empty) initEmpty()
```

Comment:
The constructors here are a bit confusing. In particular, `new Region(false)` gives an invalid region. What do you think about this:
```scala
object Region {
  // get rid of the unused `sizeHint` overload
  private def apply(blockSize: Int): Region = new Region(false, blockSize)
  
  def apply(): Region = Region(64 * 1024)

  def small(): Region = Region(8 * 1024)

  def tiny(): Region = Region(1024)
}

final class Region private (empty: Boolean, blockSize: Int) extends NativeBase() {
  // only one constructor, which is private.
  if (empty) initEmpty() else nativeCtor(RegionPool.get, blockSize)
  ...
}
```